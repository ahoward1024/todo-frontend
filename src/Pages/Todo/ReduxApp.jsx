import React from 'react';
import AddTodo from './AddTodo';
import CheckAll from './CheckAll';
import TodoList from './TodoList';
import {requestGetState} from './ReduxActions';
import {store} from './ReduxIndex';

class ReduxApp extends React.Component {

  componentDidMount() {
    store.dispatch(requestGetState());
  }

  render() {
    return (
      <div>
        <h1 className="H1-Animation">Todo</h1>
        <AddTodo />
        <CheckAll />
        <TodoList />
      </div>
    );
  }
}

export default ReduxApp;
