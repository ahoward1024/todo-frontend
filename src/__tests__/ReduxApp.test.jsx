import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import ReduxApp from '../Pages/Todo/ReduxApp';

const initialState = {
  'checkall': false,
  'todos': []
};
const mockStore = configureStore();
const store = mockStore(initialState);

describe('ReduxApp testing', () => {
  test('Create ReduxApp component', () => {
    const tree = renderer.create(<Provider store={store}>
                                  <ReduxApp />
                                 </Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
