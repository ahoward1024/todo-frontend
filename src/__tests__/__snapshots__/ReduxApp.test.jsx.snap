// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReduxApp testing Create ReduxApp component 1`] = `
<div>
  <h1
    className="H1-Animation"
  >
    Todo
  </h1>
  <div
    className="AddTodo"
  >
    <input
      className="AddTodoInput"
      onChange={[Function]}
      placeholder=""
    />
    Â 
    <button
      className="AddTodoButton"
      onClick={[Function]}
    >
      Add Todo
    </button>
  </div>
  <div
    className="CheckboxCheckAll"
  >
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
    />
    <label>
      Check All
    </label>
  </div>
  <div
    className="Todo"
  />
</div>
`;
